File: OpenCV-Python/Code/2_GUI_Test/6_Tracebar/Tracebar_draw.py
import numpy as np
import cv2

img = np.zeros((512,512,3),np.uint8)
cv2.namedWindow('image')

draw = False
mode = True
ix,iy = -1,-1

def trackbar_callback():
    pass

cv2.createTrackbar('R','image',0,255,trackbar_callback)
cv2.createTrackbar('G','image',0,255,trackbar_callback)
cv2.createTrackbar('B','image',0,255,trackbar_callback)

def mouse_event_callback(event,x,y,flags,param):
    global ix,iy,draw,mode
    r = cv2.getTrackbarPos('R','image')
    g = cv2.getTrackbarPos('G','image')
    b = cv2.getTrackbarPos('B','image')
    if event == cv2.EVENT_LBUTTONDOWN:
        draw = True
        ix,iy = x,y
    elif event == cv2.EVENT_MOUSEMOVE and flags == cv2.EVENT_LBUTTONDOWN:
        if draw == True:
            if mode == True:
                cv2.circle(img,(x,y),10,(b,g,r),-1,cv2.LINE_AA)
            elif mode == False:
                cv2.rectangle(img,(x-5,y-5),(x+5,y+5),(b,g,r),-1,cv2.LINE_AA)
    elif event == cv2.EVENT_LBUTTONUP:
        draw = False
    
cv2.setMouseCallback('image',mouse_event_callback)
while(1):
    cv2.imshow('image',img)
    key = cv2.waitKey(1)
    if key == 27:
        break
    elif key == ord('q'):
        mode =not mode
    
cv2.destroyAllWindows()

File: OpenCV-C++/Code/2_GUI_Test/tracebar_draw.cpp
#include <opencv2/opencv.hpp>
#include <iostream>

using namespace std;
using namespace cv;

bool draw_flag = false;
bool mode = false;

Mat img;

void tracebar_callback(int value, void *userdata)
{
    ;
}

void mouse_callback(int event, int x, int y, int flags, void *param)
{
    int r, g, b;
    r = getTrackbarPos("R", "img");
    g = getTrackbarPos("G", "img");
    b = getTrackbarPos("B", "img");
    if (event == EVENT_LBUTTONDOWN)
    {
        draw_flag = true;
    }
    else if ((event == EVENT_MOUSEMOVE) && (flags == EVENT_LBUTTONDOWN))
    {
        if (draw_flag == true)
        {
            if (mode == true)
            {
                circle(img, Point(x, y), 10, Scalar(b, g, r), -1, LINE_AA);
            }
            else
            {
                rectangle(img, Point(x - 5, y - 5), Point(x + 5, y + 5), Scalar(b, g, r), -1, LINE_AA);
            }
        }
    }
    else if (event == EVENT_LBUTTONUP)
    {
        draw_flag = false;
    }
}

int main()
{
    namedWindow("img");
    img = Mat::zeros(Size(640, 480), CV_8UC3);
    createTrackbar("R", "img", 0, 255, tracebar_callback);
    createTrackbar("G", "img", 0, 255, tracebar_callback);
    createTrackbar("B", "img", 0, 255, tracebar_callback);
    setMouseCallback("img", mouse_callback);
    for (;;)
    {
        imshow("img", img);
        int key = waitKey(1);
        if (key == 27)
        {
            break;
        }
    }
}